### ISSUE TITLE
Improvement: define a protocol to handle base loss and all chunked loss.

### ISSUE THREAD

**Issue by insop** [pytorch/torchtune#2226]
As an improvement, it would be great to "define a protocol `ChunkedLoss` or something having a method e.g. `def base_loss(self, *args, **kwargs)` and have all of `CEWithChunkedOutputLoss, ForwardKLWithChunkedOutputLoss, ReverseKLWithChunkedOutputLoss, SymmetricKLWithChunkedOutputLoss` inherit from that."

Code: [_compile.py](https://github.com/pytorch/torchtune/blob/d23fa93d21045ab88f6914051f11de2b9afcd9eb/torchtune/training/_compile.py#L78)

**Context:**
This is based on the discussion in PR #2094.

**Comment by joecummings** [comment]
@felipemello1 - can we do this?
